SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -249,24 +249,32 @@ void USpatialReceiver::HandleActorAuthority(Worker_AuthorityChangeOp& Op)
 		{
 			if (Op.authority == WORKER_AUTHORITY_AUTHORITATIVE)
 			{
-				Actor->Role = ROLE_Authority;
-				Actor->RemoteRole = ROLE_SimulatedProxy;
-
-				if (Actor->IsA<APlayerController>())
-				{
-					Actor->RemoteRole = ROLE_AutonomousProxy;
-				}
-				else if (APawn* Pawn = Cast<APawn>(Actor))
+				if (IsValid(NetDriver->GetActorChannelByEntityId(Op.entity_id)))
```

Comment:
Could we use `!= nullptr`?

Reply:
`IsValid` checks for null as well as PendingKill.